Mathlib/Topology/Algebra/ContinuousMonoidHom.lean

```diff
@@ -413,3 +415,289 @@ theorem locallyCompactSpace_of_hasBasis (V : ℕ → Set Y)
 end LocallyCompact
 
 end ContinuousMonoidHom
+
+end
+
+section
+
+/-!
+
+# Continuous MulEquiv
+
+This section defines the space of continuous isomorphisms between two topological groups.
+
+## Main definitions
+
+-/
+
+universe u v
+
+variable (G : Type u) [TopologicalSpace G] (H : Type v) [TopologicalSpace H]
+
+/-- The structure of two-sided continuous isomorphisms between additive groups.
+Note that both the map and its inverse have to be continuous. -/
+structure ContinuousAddEquiv [Add G] [Add H] extends AddEquiv G H , Homeomorph G H
+
+/-- The structure of two-sided continuous isomorphisms between groups.
+Note that both the map and its inverse have to be continuous. -/
+@[to_additive "The structure of two-sided continuous isomorphisms between additive groups.
+Note that both the map and its inverse have to be continuous."]
+structure ContinuousMulEquiv [Mul G] [Mul H] extends MulEquiv G H , Homeomorph G H
+
+/-- The homeomorphism induced from a two-sided continuous isomorphism of groups. -/
+add_decl_doc ContinuousMulEquiv.toHomeomorph
+
+/-- The homeomorphism induced from a two-sided continuous isomorphism additive groups. -/
+add_decl_doc ContinuousAddEquiv.toHomeomorph
+
+@[inherit_doc]
+infixl:25 " ≃ₜ* " => ContinuousMulEquiv
+
+@[inherit_doc]
+infixl:25 " ≃ₜ+ " => ContinuousAddEquiv
+
+section
+
+namespace ContinuousMulEquiv
+
+variable {M N : Type*} [TopologicalSpace M] [TopologicalSpace N] [Mul M] [Mul N]
+
+section coe
+
+@[to_additive]
+instance : EquivLike (M ≃ₜ* N) M N where
+  coe f := f.toFun
+  inv f := f.invFun
+  left_inv f := f.left_inv
+  right_inv f := f.right_inv
+  coe_injective' f g h₁ h₂ := by
+    cases f
+    cases g
+    congr
+    exact MulEquiv.ext_iff.mpr (congrFun h₁)
+
+@[to_additive]
+instance : MulEquivClass (M ≃ₜ* N) M N where
+  map_mul f := f.map_mul'
+
+@[to_additive]
+instance : HomeomorphClass (M ≃ₜ* N) M N where
+  map_continuous f := f.continuous_toFun
+  inv_continuous f := f.continuous_invFun
+
+@[to_additive] -- shortcut instance that doesn't generate any subgoals
+instance : CoeFun (M ≃ₜ* N) fun _ ↦ M → N where
+  coe f := f
+
+/-- Two continuous multiplicative isomorphisms agree if they are defined by the
+same underlying function. -/
+@[to_additive (attr := ext)
+  "Two continuous additive isomorphisms agree if they are defined by the same underlying function."]
+theorem ext {f g : M ≃ₜ* N} (h : ∀ x, f x = g x) : f = g :=
+  DFunLike.ext f g h
+
+@[to_additive]
+protected theorem congr_arg {f : M ≃ₜ* N} {x x' : M} : x = x' → f x = f x' :=
+  DFunLike.congr_arg f
+
+@[to_additive]
+protected theorem congr_fun {f g : M ≃ₜ* N} (h : f = g) (x : M) : f x = g x :=
+  DFunLike.congr_fun h x
+
+@[to_additive (attr := simp)]
+theorem coe_mk (f : M ≃* N) (hf1 hf2) : (mk f hf1 hf2 : M → N) = f := rfl
+
+@[to_additive]
+theorem toEquiv_eq_coe (f : M ≃ₜ* N) : f.toEquiv = f :=
+  rfl
+
+@[to_additive, simp]
```

Comment:
```suggestion
@[to_additive (attr := simp)]
```
I think what you have is wrong -- you don't get the simp attribute on the additivised lemma.

Reply:
Sorry that I am not that familiar with these, thank you.